pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/RealtimeSegmentImpl.java

```diff
@@ -71,13 +70,10 @@
   private final Map<String, RealtimeInvertedIndexReader> invertedIndexMap = new HashMap<>();
   private final Map<String, MutableDictionary> dictionaryMap = new HashMap<>();
 
-  private AtomicInteger docIdGenerator;
   private String outgoingTimeColumnName;
   private TimeGranularitySpec outgoingGranularitySpec;
 
-  private int docIdSearchableOffset = -1;
-  private int numDocsIndexed = 0;
-  private int numSuccessIndexed = 0;
+  private transient int _numDocsIndexed = 0;
```

Comment:
volatilie? Why transient?

Reply:
Good catch! Should be volatile, forgive my poor English...